File: src/fetch.php
<?php
$queries = ['Matematika SMA', 'Fisika SMA', 'Kimia SMA', 'Biologi Sel', 'Sejarah Kemerdekaan', 'Bahasa Indonesia SMA', 'Bahasa Inggris SMA', 'Ekonomi SMA'];
$results = [];

foreach ($queries as $q) {
    $url = 'https://www.youtube.com/results?search_query=' . urlencode($q);
    $html = @file_get_contents($url, false, stream_context_create(['http' => ['header' => 'User-Agent: Mozilla/5.0']]));
    if ($html) {
        preg_match_all('/"videoId":"([^"]{11})"/', $html, $matches);
        $ids = array_unique($matches[1]);
        $results[$q] = array_slice($ids, 0, 2);
    }
}
echo json_encode($results, JSON_PRETTY_PRINT);

File: src/fetch.py
import urllib.request
import re
import json

queries = ['Matematika SMA', 'Fisika SMA', 'Kimia SMA', 'Biologi Sel', 'Sejarah Kemerdekaan', 'Bahasa Indonesia SMA', 'Bahasa Inggris SMA']
results = {}

for q in queries:
    url = 'https://www.youtube.com/results?search_query=' + urllib.parse.quote(q)
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        html = urllib.request.urlopen(req).read().decode()
        ids = re.findall(r'"videoId":"(.*?)"', html)
        # filter out very short strings just in case
        ids = [i for i in ids if len(i) == 11]
        results[q] = list(set(ids))[:2]
    except Exception as e:
        results[q] = str(e)

print(json.dumps(results, indent=2))
